llama-index-integrations/postprocessor/llama-index-postprocessor-rankllm-rerank/llama_index/postprocessor/rankllm_rerank/base.py

```diff
@@ -0,0 +1,79 @@
+from typing import Any, List, Optional
+
+from llama_index.core.bridge.pydantic import Field, PrivateAttr
+from llama_index.core.postprocessor.types import BaseNodePostprocessor
+from llama_index.core.schema import NodeWithScore, QueryBundle
+
+
+class RankLLMRerank(BaseNodePostprocessor):
+    """RankLLM-based reranker."""
+    top_n: int = Field(default=5, description="Top N nodes to return from reranking.")
+    model: str = Field(default="zephyr", description="Reranker model name.")
+    with_retrieval: bool = Field(default=False, description="Perform retrieval before reranking.")
+    _model: Any = PrivateAttr()
+    _result: Any = PrivateAttr()
+    _retriever: Any = PrivateAttr()
+    
+    def __init__(
+        self,
+        top_n: int = 10,
+        with_retrieval: bool = False,
+        model: str = "zephyr",
+    ):
+        try: 
+            from rank_llm.result import Result 
+            self._result = Result
+
+
+            if model == "vicuna":
+                from rank_llm.rerank.vicuna_reranker import VicunaReranker
+                self._model = VicunaReranker()
+            else:
```

Comment:
should this be an `Enum` or `Literal` type instead, and we raise an error if user enters a different model name than `vicuna` or `zephyr`?